### PR TITLE
Add Visualizations

### PR DESCRIPTION
**Pull request by sophiebpeneva**

### CODE REVIEW

**Review by peterborocz — CHANGES_REQUESTED** [review]

File: frontend/src/app/app.component.ts [hunk 1]
```diff
@@ -1,26 +1,62 @@
...
+
+  public uploadFile(file): void {
+    this.api.uploadFile(file).subscribe(response => {
+      this.getData();
+    });
   }
 
-  public uploadFile(): void {
-    this.api.uploadFile(this.files).subscribe(response => {});
+  signalChildFOrChartUpdates(changes): void {
```
[the remark this answers is not in the document]
Comment (sophiebpeneva) [reply frontend/src/app/app.component.ts]: done

File: frontend/src/app/api.service.ts [hunk 2]
```diff
@@ -12,14 +12,59 @@ export class ApiService {
   constructor(private http: HttpClient) {
   }
 
+  getSegmentedData(filename, page = 0, perPage = 8, filters = null): Observable<object> {
```
[the remark this answers is not in the document]
Comment (sophiebpeneva) [reply frontend/src/app/api.service.ts:24]: done

File: frontend/src/app/api.service.ts [hunk 3]
```diff
@@ -12,14 +12,59 @@ export class ApiService {
...
+
+    let params = new HttpParams()
+      .set('page', page.toString())
+      .set('per_page', perPage.toString());
+    params = this.buildFilter(filters, params);
+    return this.http.get(environment.apiRest + 'compare/' + filename + '/' + filename2, {params});
+  }
+
+  getErrorData(filename, filename2): Observable<object> {
+    return this.http.get(environment.apiRest + 'error/' + filename + '/' + filename2);
```
[the remark this answers is not in the document]
Comment (sophiebpeneva) [reply frontend/src/app/api.service.ts]: I removed it as it should not have been in this commit. I will keep in mind that I have to rename it once I do push it.

File: frontend/src/app/api.service.ts [hunk 4]
```diff
@@ -12,14 +12,59 @@ export class ApiService {
...
+    return this.http.get(environment.apiRest + 'data/' + filename, {params});
+  }
+
+  getComparisonData(filename, filename2, page = 0, perPage = 8, filters = null): Observable<object> {
+
+    let params = new HttpParams()
+      .set('page', page.toString())
+      .set('per_page', perPage.toString());
+    params = this.buildFilter(filters, params);
+    return this.http.get(environment.apiRest + 'compare/' + filename + '/' + filename2, {params});
```
[the remark this answers is not in the document]
Comment (sophiebpeneva) [reply frontend/src/app/api.service.ts]: removed this method from the commit

File: frontend/src/app/api.service.ts [hunk 5]
```diff
@@ -12,14 +12,59 @@ export class ApiService {
   constructor(private http: HttpClient) {
   }
 
+  getSegmentedData(filename, page = 0, perPage = 8, filters = null): Observable<object> {
+
```
[the remark this answers is not in the document]
Comment (sophiebpeneva) [reply frontend/src/app/api.service.ts]: done

**Review by peterborocz — APPROVED** [review]

**Pull request merged**